azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -417,21 +448,36 @@ def supported(self):
             return self._cgroups_supported
 
         def enabled(self):
-            return self._cgroups_enabled
+            return self._agent_cgroups_enabled
+
+        def ext_enabled(self):
+            return self._ext_cgroups_enabled
 
         def enable(self):
             if not self.supported():
                 raise CGroupsException("Attempted to enable cgroups, but they are not supported on the current platform")
-            self._cgroups_enabled = True
+            self._agent_cgroups_enabled = True
+            self._ext_cgroups_enabled = True
             self.__set_cpu_quota(conf.get_agent_cpu_quota())
 
-        def disable(self, reason):
-            self._cgroups_enabled = False
+        def disable(self, reason, telemetry_name = None):
+            # Todo: disable/reset extension when ext quotas introduced
+            if telemetry_name is None:                 # disable all
+                self._agent_cgroups_enabled = False
+                self._ext_cgroups_enabled = False
+                self.__reset_cpu_quota()
```

Comment:
can we rename "__reset_cpu_quota" to "__reset_agent_cpu_quota"?

Reply:
addressed